devtools/cli/src/main/java/io/quarkus/cli/Create.java

```diff
@@ -0,0 +1,140 @@
+public class Create extends BaseSubCommand implements Callable<Integer> {
+
+    @CommandLine.Option(names = { "-g",
+            "--group-id" }, order = 1, paramLabel = "GROUP-ID", description = "The groupId for project")
+    String groupId = "org.acme";
+
+    @CommandLine.Option(names = { "-a",
+            "--artifact-id" }, order = 2, paramLabel = "ARTIFACT-ID", description = "The artifactId for project")
+    String artifactId = "my-project";
+
+    @CommandLine.Option(names = { "-v",
+            "--version" }, order = 3, paramLabel = "VERSION", description = "The version for project")
+    String version = "1.0-SNAPSHOT";
+
+    @CommandLine.Option(names = { "-0",
+            "--empty" }, order = 4, description = "Generate without example code.")
+    boolean empty = false;
+
+    @CommandLine.ArgGroup()
+    TargetBuildTool targetBuildTool = new TargetBuildTool();
+
+    static class TargetBuildTool {
+        @CommandLine.Option(names = { "--maven" }, order = 5, description = "Create a Maven project. (default)")
+        boolean maven = false;
+
+        @CommandLine.Option(names = { "--gradle" }, order = 6, description = "Create a Gradle project.")
+        boolean gradle = false;
+
+        @CommandLine.Option(names = {
+                "--grade-kotlin-dsl" }, order = 7, description = "Create a Gradle Kotlin DSL project.")
+        boolean gradleKotlinDsl = false;
+    }
+
+    @CommandLine.ArgGroup()
+    TargetLanguage language = new TargetLanguage();
+
+    static class TargetLanguage {
+        @CommandLine.Option(names = {
+                "--java" }, order = 8, description = "Generate Java examples. (default)")
+        boolean java = false;
+
+        @CommandLine.Option(names = {
+                "--kotlin" }, order = 9, description = "Generate Kotlin examples.")
+        boolean kotlin = false;
+
+        @CommandLine.Option(names = {
+                "--scala" }, order = 10, description = "Generate Scala examples.")
+        boolean scala = false;
+    }
+
+    @CommandLine.Parameters(arity = "0..1", paramLabel = "EXTENSION", description = "extension to add to project")
+    Set<String> extensions;
+
+    @Override
+    public Integer call() throws Exception {
+        try {
+            File projectDirectory = new File(System.getProperty("user.dir"));
+
+            File projectRoot = new File(projectDirectory.getAbsoluteFile(), artifactId);
+            if (projectRoot.exists()) {
+                err().println("Unable to create the project, " +
+                        "the directory " + projectRoot.getAbsolutePath() + " already exists");
+                return CommandLine.ExitCode.SOFTWARE;
+            }
+
+            SourceType sourceType = SourceType.JAVA;
+            if (targetBuildTool.gradleKotlinDsl) {
+                sourceType = SourceType.KOTLIN;
+                if (extensions == null)
+                    extensions = new HashSet<>();
```

Comment:
you could do it only once I guess

Reply:
it saves an object creation when its --java and no extensions.